compiler/src/dmd/aggregate.d

```diff
@@ -829,7 +830,11 @@ public uint placeField(ref uint nextoffset, uint memsize, uint memalignsize,
     bool overflow;
     const sz = addu(memsize, actualAlignment, overflow);
     addu(ofs, sz, overflow);
-    if (overflow) assert(0);
+    if (overflow)
+    {
+        error(loc, "integer overflow adding %u + %u + %u when placing field in aggregate", memsize, actualAlignment, ofs);
```

Comment:
While this error message correctly describes the bit of internal compiler logic that triggered it, it reads terribly when pairing it with the code that caused it.

The reason for why the code is rejected should be diagnosed to the user instead. That is, the size of the type exceeds more than half the address space or some max object size limit.

Reply:
ok